packages/sdk/src/PreVerificationGasCalculator.ts

```diff
@@ -0,0 +1,114 @@
+import { encodeUserOp, packUserOp, UserOperation } from '@account-abstraction/utils'
+import { arrayify, hexlify } from 'ethers/lib/utils'
+
+export interface PreVerificationGasCalculatorConfig {
+  /**
+   * Gas overhead is added to entire 'handleOp' bundle.
+   */
+  readonly fixedGasOverhead: number
+  /**
+   * Gas overhead per UserOperation is added on top of the above fixed per-bundle.
+   */
+  readonly perUserOpGasOverhead: number
+  /**
+   * Gas overhead per single "word" (32 bytes) of an ABI-encoding of the UserOperation.
+   */
+  readonly perUserOpWordGasOverhead: number
+  /**
+   * The gas cost of a single zero byte an ABI-encoding of the UserOperation.
+   */
+  readonly zeroByteGasCost: number
+  /**
+   * The gas cost of a single zero byte an ABI-encoding of the UserOperation.
+   */
+  readonly nonZeroByteGasCost: number
+  /**
+   * The expected average size of a bundle in current network conditions.
+   * This value is used to split the bundle gas overhead between all ops.
+   */
+  readonly expectedBundleSize: number
+  /**
+   * The size of the dummy 'signature' parameter to be used during estimation.
+   */
+  readonly estimationSignatureSize: number
```

Comment:
I wonder if we should estimate signature size at all: there shouldn't be a "standard" sig size. The wallet should fill it in.
Same for paymaster: if there isn't a pmd, we should estimate based on no pmd, not provide a "hint" that there is a "suggested" value for it.

Reply:
This does not make sense to me. There absolutely is no "standard" sig size and this configuration parameter is exactly the result of that. In order to estimate the `preVerificationGas` we must know what the size of the signature will be in the future.
When the wallet comes to a point where it needs to **estimate** the amount of `preVerificationGas`, if it did not take into account the `signature` size or the `paymasterData` size, it would calculate a lower than necessary `preVerificationGas` and its UserOps would be rejected.

**NOTE** that the "validation" step absolutely does not rely on `estimation`-related configuration parameters and relies on the real value.

If you still disagree please elaborate how you expect `preVerificationGas` to work alternatively.
Thanks.